<head>
	<title>16S Results</title>
	<link rel="stylesheet" type="text/css" href="mystyle.css">
	<link rel="shortcut icon" type="image/x-icon" href="images/favicon.ico">
	<link rel="icon" type="image/x-icon" href="images/favicon.ico">

</head>


<h1 style="text-align: center">16S Results</h1>

Here is an explanation of what you can expect back from a 16S analysis. 16S projects are usually highly specific, so these may not apply to your project, or what you want may not be listed here. I can do more than is shown, just ask.

<h3>Alpha Diversity</h3>
This is a measure of how diverse a given sample is.

I usually provide Shannon and Simpson diversities. I can easily include any of the following:
<ul>
	<li>Observed</li>
	<li>Chao1</li>
	<li>ACE</li>
	<li>Shannon</li>
	<li>Simpson</li>
	<li>InvSimpson</li> 
	<li>Fisher</li>
</ul>
I can do others if there is a specific need.

<img src='images/alpha_by_day.png' alt='alpha image'>

<h3>Beta Diversity</h3>

Beta diversity measures the distance between two samples. Some beta measures are:
	
<div id="multicolList">
<ul>
	<li>unifrac</li>
	<li>wunifrac</li>
	<li>dpcoa</li>
	<li>jsd</li>
	<li>manhattan</li> 
	<li>euclidean</li>
	<li>canberra</li>
	<li>bray</li>
	<li>kulczynski</li>
	<li>jaccard</li>
	<li>gower</li>
	<li>altGower</li>
	<li>morisita</li>
	<li>horn</li>
	<li>mountford</li>
	<li>raup</li>
	<li>binomial</li>
	<li>chao</li>
	<li>cao</li>
	<li>w</li>
	<li>-1</li>
	<li>c</li>
	<li>wb</li>
	<li>r</li>
	<li>I</li>
	<li>e</li>
	<li>t</li>
	<li>me</li>
	<li>j</li>
	<li>sor</li>
	<li>m</li>
	<li>-2</li>
	<li>co</li>
	<li>cc</li>
	<li>g</li>
	<li>-3</li>
	<li>l</li>
	<li>19</li>
	<li>hk</li>
	<li>rlb</li>
	<li>sim</li>
	<li>gl</li>
	<li>z</li>
	<li>maximum</li>
	<li>binary</li>
	<li>minkowski</li>
</ul>
</div>


I generally do not return a matrix of distances, but I can if you like. 
What I do is use the distances calculated to do ordinations and to calculate differences between groups. 
<h4>Vegan Plots</h4>
These are produced by the R package vegan, and show if two groups of samples are different based on a particular distance metric.

<img src='images/vegan1.png' alt='vegan image'>



<h3>Ordinations</h3>
Lots of different ordinations are available. I gennerally just do a PCoA based on a couple of different metrics.

I also create an interactive PCoA, which you can play with in your browser, using the emperor plug in.
Example <a href="emperor/emp.html">here</a>.


<h3>Differential Analysis</h3>
16S sequencing produces count data, so it is entirely reasonable to use tools built for RNA-seq to determine if there are individual taxa which are significantly different between groups. Specifically, I use DESeq2. You can find information on the DESeq output <a href="https://nielinfante.github.io/RNAseq/">here.</a> 
One extra figure I produce for 16S DESeq analysis is an overview, which shows a quick image of significantly different taxa.

<img src='images/deseq_overview.png' alt='overview'>

<h3>Taxa Bar Charts</h3>
<img src='images/table.png' alt='barplot'>

<h3>Taxonomic Trees</h3>
<img src='images/tree.png' alt='Taxonomic Tree'>


<h3>Data Returned</h3>
The data that can be returned include:

<ul>
	<li>PhyloseqObject.rds  -  R data object that can be used to recreate all the following data files</li>
	<li>Phyloseq_Filtered.rds  -  Same as the above, after various filters are applied</li>
	<li>OTU Table  -  A text file listing all taxa and the count of each for eachsample</li>
	<li>Taxa Table  -  A text file listing ASVs and their taxonomic classification</li>
</ul>













